### PR TITLE
README.md: shorten and add Discord link

### PR DESCRIPTION
**Pull request by steve-s**
1) Adds the new Discord server link to README.md
2) I think we duplicate a lot of things in readme that are also in the documentation. My suggestion is to make the readme short and link the documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,73 +2,51 @@
 
 [![Build](https://github.com/hpyproject/hpy/actions/workflows/ci.yml/badge.svg)](https://github.com/hpyproject/hpy/actions/workflows/ci.yml)
 [![Documentation](https://readthedocs.org/projects/hpy/badge/)](https://hpy.readthedocs.io/)
+[![Join the discord server at https://discord.gg/xSzxUbPkTQ](https://img.shields.io/discord/1077164940906995813.svg?color=7389D8&labelColor=6A7EC2&logo=discord&logoColor=ffffff&style=flat-square)](https://discord.gg/xSzxUbPkTQ)
 
-**IRC channel** : #hpy on [libera.chat](https://libera.chat)
-
+**Documentation**: [hpy.readthedocs.io](https://hpy.readthedocs.io/) \
```
Comment (mattip) [inline README.md:7]: Maybe https://hpyproject.org or https://docs.hpyproject.org/en/latest/overview.html?

**Review by mattip — COMMENTED** [review]
This is great, thanks! One small nit about the documentation link.

File: README.md (hunk 1 above)
Comment (steve-s) [reply README.md:7]: Good point. Maybe it should be "Website: hpyproject.org" linking to https://hpyproject.org/?

Comment (mattip) [reply README.md:7]: gr8

**Pull request merged**